### ISSUE TITLE
Search state exploded on a list

### ISSUE THREAD

**Issue by thesamet** [scalameta/scalafmt#2022]
```
version = "2.5.3"
```

```scala
object X {
  def f(s: String): String = ???
}

object ProtobufGenerator {
  def foobar(x: Int) =
    X.f("foo") ::
      X.f("foo") ::
      X.f("foo") ::
      X.f("foo") ::
      X.f("foo") ::
      X.f("foo") ::
      X.f("foo") ::
      X.f("foo") ::
      X.f("foo") ::
      X.f(
        s"very long string very long string very long string very long string very long string very long string"
      ) :: Nil
}
```

Upon formatting I am getting
```
[info] Formatting 1 Scala sources...
[error] Search state exploded on '::∙X[274:282]', line 14: /tmp/assoc/src/main/scala/Hello.scala
[error] (Compile / scalafmt) Search state exploded on '::∙X[274:282]', line 14: /tmp/assoc/src/main/scala/Hello.scala
```

I am aware that there are [known limitations related to deeply nested definitions](https://scalameta.org/scalafmt/docs/known-issues.html#deeply-nested-code). Given that this list has only 11 elements, and the problem only happens when there's a long line involved, I wonder if this still can be fixed.

The workaround I've been using was to break the long string into two strings on separate lines and concatenate.

I am curious if this happens since scalafmt goes through different options to join the lines. If so, it would be nice to have an option to provide a list of tokens (such as `::`) that scalafmt should always leave at the end of the line.

**Comment by kitbellew** [comment]
@thesamet since you didn't provide any configuration overrides, i will assume you are using the default configuration.

however, in that case, i am befuddled as what you are describing shouldn't happen; you have a long infix expression, and by default the formatter doesn't mess with line break around infix operators.

are you using any configuration settings other than version?

**Comment by thesamet** [comment]
Thanks for the quick reply. Confirming that `version` is the only key. I am using sbt-scalafmt 2.4.0.

**Comment by thesamet** [comment]
I am not sure if this was intentional, but the issue is resolved in 2.6.0 - feel free to close.
